### PR TITLE
Navbar Styling Updates. 404 Page Styling Updates.

### PR DESCRIPTION
**Pull request by tackc**
#### Please check if the PR fulfills these requirements

- [x] The changes don't come from your master branch. ([Source](https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/)) 

---

#### What kind of change does this PR introduce? (bug fix, feature, docs update, ...)

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

#### What is the current behavior? (You can also link to an open issue here)
- Navbar was not mobile responsive
- `ErrorPage` had layout issue causing page to be too tall on mobile and displayed white space below yellow bar at bottom

#### What is the new behavior? (if this is a feature change)
- Both pages are now mobile responsive
- Hamburger menu shows up on small screens
- Fixed styling issue on the `HomePage` where components were overlapping at smaller screen sizes
- Fixed issue in `DonationCard` component where it was displaying "NaN" instead of 0% (when applicable)
- Updated `Icon` component to allow className to be passed down from "parent"

#### Images (before/ after screenshots, interaction GIFs, ...)

Screenshot 2023-11-14 at 9 14 36 AM

Screenshot 2023-11-14 at 9 14 54 AM

Screenshot 2023-11-14 at 9 22 12 AM

---

#### TODO:
- The route for `/clients` needs to be created. Currently, that link leads users to the 404 page

### CODE REVIEW

File: src/Pages/ErrorPage/index.js [hunk 1]
```diff
@@ -25,25 +25,25 @@ function ErrorPage() {
...
+    <div className={styles.container}>
       {isAuthenticated() && <Navbar />}
+
       <div className={styles.body}>
-        <div className={styles.leftBody}>
+        <div className={styles.bodyContent}>
           <h1 className={styles.heading}>404</h1>
           <p className={styles.subhead}>Looks like this page went bananas.</p>
           <div className={styles.buttonContainer}>
             <Button type="submit" variant="buttonSecondary" text="Go back" action={goBack} />
```
Comment (renburnett) [inline src/Pages/ErrorPage/index.js:36]: @tackc looks like on the figma example they removed the button borders with `border: none`

Comment (renburnett) [reply src/Pages/ErrorPage/index.js:36]: image

File: src/Components/Navbar/index.js [hunk 2]
```diff
@@ -1,71 +1,63 @@
...
+          </NavItem>
+
+          <UncontrolledDropdown nav inNavbar className={styles.hideMobile}>
+            <DropdownToggle
+              nav
+              caret
+              className={`${styles.dropdownToggleMobile} + ${styles.navbarText}`}
+            >
+              <Icon name="tasks" size={32} className={styles.navIcon} />
+              <NavbarText className={styles.displayMobile}>Tasks</NavbarText>
```
Comment (renburnett) [inline src/Components/Navbar/index.js:58]: @tackc its kinda nit-picky but maybe these subnav links should span the full width of their parent container? if not, no worries ✔️ 
image

**Review by renburnett — APPROVED** [review]
looks great!! two small potential changes, not 100% needed though, feel free to merge!

File: src/Components/Navbar/index.js (hunk 2 above)
Comment (tackc) [reply src/Components/Navbar/index.js:58]: Good catch! I also added the same hover styling to the `Notifications` and `Profile` links!

File: src/Pages/ErrorPage/index.js (hunk 1 above)
Comment (tackc) [reply src/Pages/ErrorPage/index.js:36]: Sure enough! I totally missed that! Thanks...it's resolved 😀

**Pull request merged**